### PR TITLE
Enable editing upon row click

### PR DESCRIPTION
**Pull request by klassm**
## Description

The idea is to enable starting the row edit mode upon clicking the row.

### CODE REVIEW

**Comment by klassm** [comment]
This PR should become green as soon as https://github.com/material-table-core/core/pull/783 has been merged.

File: src/components/MTableBodyRow/index.js [hunk 1]
```diff
@@ -49,17 +50,24 @@ function MTableBodyRow({ forwardedRef, ...props }) {
...
+          panel = panel[panelIndex || 0];
+          if (typeof panel === 'function') {
+            panel = panel(data);
+          }
+          panel = panel.render;
         }
-        panel = panel.render;
+        onToggleDetailPanel(path, panel);
+      },
+      () => {
```
Comment (Domino987) [inline src/components/MTableBodyRow/index.js]: Can we make this into an action obejct to make it more clearer? The two callbacks?

File: types/index.d.ts [hunk 2]
```diff
@@ -79,7 +79,8 @@ export interface MaterialTableProps<RowData extends object> {
   onRowClick?: (
     event?: React.MouseEvent,
     rowData?: RowData,
-    toggleDetailPanel?: (panelIndex?: number) => void
+    toggleDetailPanel?: (panelIndex?: number) => void,
+    enableEditMode?: () => void
```
Comment (Domino987) [inline types/index.d.ts]: those would need to be updated as well

**Comment by klassm** [comment]
The callbacks are now an actions object. Beware: This will be breaking now.

**Pull request merged**